### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by mstrother** [Azure/DotNetty#213]
I moved a project from a 'roll your own' socket solution to DotNetty. The performance increase blew me away. Netty's Java documentation helped but it would have be useful to have native documentation. What did help a bit was the documentation and code for the Azure-IoT-Protocol gateway. [This page](https://github.com/Azure/azure-iot-protocol-gateway/blob/master/docs/DeveloperGuide.md), has a section called **Component Structure** which runs down the basic architecture of DotNetty.  That would be  useful, almost verbatim, in the DotNetty. It would have got me up to speed faster. It could serve as a starting ground for future, more in depth, documentation. 

Is there any support on adding that text to the DotNetty README.md? I'd be willing to take it on. It would be a quick task.

**Comment by me66ccff** [comment]
Dotnetty really need a documentation , I rely on reading java netty documents and reading dotnetty source code to learn this framework. Not a good experience.

Just do it~
Cheers~

**Comment by StormHub** [comment]
@me66ccff the wiki page is open :D

**Comment by introspection3** [comment]
Dotnetty really need a documentation +++

**Comment by nshCore** [comment]
Yes this is +1 from me as well. It's all good saying read the netty docs like here

https://github.com/Azure/DotNetty/issues/135#issuecomment-228128385

but that really isn't applicable apart from high level concepts.

**Comment by monsieurleberre** [comment]
Yes please!

**Comment by Kasperki** [comment]
yes please

**Comment by introspection3** [comment]
DotNet has been prejudiced by Javaer for too many years,Now we must have a document to develop high
performence soft ware...

**Comment by bugproof** [comment]
Don't expect many people to use this library without any documentation or "Quickstart"
